rest-api/src/main/java/io/quarkus/qe/services/RepositoryService.java

```diff
@@ -57,8 +84,17 @@ public Repository findByRepoUrl(String repoUrl) throws RepositoryNotFoundExcepti
         return repositoryMarshaller.fromEntity(entity);
     }
 
-    public List<Repository> findAll() {
-        return RepositoryEntity.<RepositoryEntity> findAll()
+    public List<Repository> findAll(List<String> versions) {
+        StringBuilder sql = new StringBuilder();
+        String quarkusVersionParamName = "quarkusVersion";
+        sql.append("select distinct repo from repository repo\t");
+        Map<String, Object> versionsParam = getQuarkusVersionAsQueryParam(quarkusVersionParamName, versions);
+
+        if (!versionsParam.isEmpty()) {
+            sql.append("where quarkusVersion IN :" + quarkusVersionParamName);
+        }
+
+        return RepositoryEntity.<RepositoryEntity> find(sql.toString(), versionsParam)
```

Comment:
Can you refactor this piece of code to use Parameters instead of maps? https://quarkus.io/guides/hibernate-orm-panache#query-parameters